dask_kubernetes/operator/daskcluster.py

```diff
@@ -237,6 +239,9 @@ async def daskworkergroup_update(spec, name, namespace, logger, **kwargs):
             )
         logger.info(f"Scaled worker group {name} up to {spec['replicas']} workers.")
     if workers_needed < 0:
+        # with rpc(f"tcp://{scheduler_name}.{namespace}:8786") as scheduler:
+        #     scheduler_info = scheduler
```

Comment:
So this should be fine. The `scheduler` here would be an RPC representation of the scheduler where you can asynchronously invoke any of the [exposed handlers](https://github.com/dask/distributed/blob/a364d426b0565f5cce17bf814a576a8fd84206af/distributed/scheduler.py#L3893-L3973).

```python
from distributed.core import rpc
with rpc("localhost:8786") as scheduler:
    worker_ids = await scheduler.workers_to_close(n=5)
```

The above code tells the scheduler that you want to remove `5` workers. The scheduler would select 5 workers to remove, move any futures they are holding onto different workers and then return a list of the IDs of the workers that are ready for deletion. We can then use that list to select which pods to delete.

Reply:
In `build_worker_pod_spec` we probaby want to ensure that we set the `--name` flag on each worker with a name that matches the pod name, because this is the ID that the `workers_to_close` method will give us. Should make it easier to delete the right pods that way.